log-collector/src/main/scala/io/phdata/pulse/logcollector/KerberosUtil.scala

```diff
@@ -0,0 +1,46 @@
+/* Copyright 2018 phData Inc. */
+
+package io.phdata.pulse.logcollector
+
+import java.security.PrivilegedAction
+import java.util.concurrent.TimeUnit
+
+import com.typesafe.scalalogging.LazyLogging
+import javax.security.auth.Subject
+import javax.security.auth.login.LoginContext
+import monix.execution.Cancelable
+import monix.execution.Scheduler.{ global => scheduler }
+
+object KerberosUtil extends LazyLogging {
+
+  var lc = new LoginContext("Client")
+
+  def scheduledLogin(initialDelay: Long, delay: Long, timeUnit: TimeUnit): Cancelable = {
+    val runnableLogin = new Runnable {
+      def run(): Unit = {
+        getSubject.foreach(_ => logout())
+        lc = new LoginContext("Client")
+        login()
+      }
+    }
+    scheduler.scheduleWithFixedDelay(initialDelay, delay, timeUnit, runnableLogin)
+  }
+
+  def run[F](function: => Any): Any =
+    Subject.doAs(lc.getSubject, new PrivilegedAction[Any]() {
+      override def run: Any = function
+    })
+
+  def login(): Unit =
+    lc.login()
+
+  def logout(): Unit =
+    lc.logout()
+
+  def getSubject: Option[String] =
+    if (lc.getSubject == null) {
```

Comment:
When does this happen? I think we just want to throw an exception here or let it die

Reply:
If we need this I think you can just use `Option(lc.getSubject)`